hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-replication-core/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/IncompatibleMutationAdapter.java

```diff
@@ -99,6 +114,21 @@ private boolean isValidDelete(Cell delete) {
     List<Cell> returnedCells = new ArrayList<>(cellsToAdapt.size());
     for (int index = 0; index < cellsToAdapt.size(); index++) {
       Cell cell = cellsToAdapt.get(index);
+      // check whether there is timestamp overflow from HBase -> CBT and make sure
+      // it does clash with valid delete which require the timestmap to be
+      // HConstants.LATEST_TIMESTAMP,
+      if (cell.getTimestamp() >= HBASE_EFFECTIVE_MAX_TIMESTAMP
+          && cell.getTimestamp() != HConstants.LATEST_TIMESTAMP) {
+        incrementTimestampOverflowMutations();
+        LOG.error(
```

Comment:
We should not log here, if customer is doing reverse timestamps, this can be all the PUTs. If you want to log it. log it in trace mode. 

So you can log this and future puts in trace, users can enable it in test if they want to. Be very prescriptive that DO NOT enable trace logging in prod.